src/rpdk/guard_rail/core/stateful.py

```diff
@@ -89,16 +91,59 @@ class DIFFKEYS:
 }
 
 
+def remove_key(document: Dict[str, Any], key: str):
+    """removes only top level property"""
+    cloned_document = deepcopy(document)
+    if key in cloned_document:
+        del cloned_document[key]
+    return cloned_document
+
+
+def get_value(document: Dict[str, Any], key: str):
+    """retrieves only top level property"""
+    return document.get(key)
+
+
 def schema_diff(previous_json: Dict[str, Any], current_json: Dict[str, Any]):
     """schema diff function to get formatted schema diff from deep diff"""
-    deep_diff = DeepDiff(
-        resolve_schema(previous_json),
-        resolve_schema(current_json),
+
+    previous_schema = resolve_schema(previous_json)
+    current_schema = resolve_schema(current_json)
+
+    previous_schema_no_primary_id = remove_key(previous_schema, "primaryIdentifier")
+    current_schema_no_primary_id = remove_key(current_schema, "primaryIdentifier")
+
+    previous_schema_only_primary_id = {
+        "primaryIdentifier": get_value(previous_schema, "primaryIdentifier")
+    }
+    current_schema_only_primary_id = {
+        "primaryIdentifier": get_value(current_schema, "primaryIdentifier")
+    }
+
+    deep_diff_no_primary_id = DeepDiff(
+        previous_schema_no_primary_id,
+        current_schema_no_primary_id,
         ignore_order=True,
         verbose_level=2,
     )
-    print(json.dumps(_translate_meta_diff(deep_diff.to_dict())))
-    return _translate_meta_diff(deep_diff.to_dict())
+
+    deep_diff_only_primary_id = DeepDiff(
+        previous_schema_only_primary_id,
+        current_schema_only_primary_id,
+        ignore_order=False,
+        verbose_level=2,
+    )
+
+    meta_diff_no_primary_id = _translate_meta_diff(deep_diff_no_primary_id.to_dict())
+    meta_diff_only_primary_id = _translate_meta_diff(
+        deep_diff_only_primary_id.to_dict()
+    )
+    cumulative_meta_diff = {**meta_diff_no_primary_id, **meta_diff_only_primary_id}
+
+    print("[GENERATED DIFF BETWEEN SCHEMAS]")
```

Comment:
`console.print()` instead of `print()`?

Reply:
right, this is the rich output with a lot of built-in prettifying features

Reply:
Sorry, I was not clear. I meant to say that we should stick with `console.print()` instead of printing using `print()` as well.

Code:
`print("[GENERATED DIFF BETWEEN SCHEMAS]")`

Suggestion:
`console.print("[GENERATED DIFF BETWEEN SCHEMAS]")`